enhancements/installer/openstack-customer-provided-subnets.md

```diff
@@ -0,0 +1,151 @@
+
+To accomplish this, our infrastructure needs an OpenStack account access that can be separated into "networking infrastructure" versus "application". Networking infrastructure includes elements related to the networks (networks, subnets, routing tables, internet gateways, NAT, VPN). Application items include OpenStack resources directly touching nodes within the cluster (LBs, security groups, Swift containers, nodes). This is a typical separation of responsibilities among IT groups and deployment/provisioning pipelines within large companies.
+
+## Motivation
+
+### Goals
+As an administrator, I would like to create or reuse my own networks, subnets and routing schemes that I can deploy OpenShift to.
+
+### Non-Goals
+There are no changes in expectations regarding publicly addressable parts of the cluster.
+
+## Proposal
+- Installer allow users to provide a list of subnets that should be used for the cluster. Since there is an expectation that networking is being shared, the installer cannot modify the networking setup, but changes required to the shared resources like Tags that do not affect the behavior for other tenants of the network will be made.
+- The installer validates a set of bare minimum assumptions about the networking setup
+- Non-networking infrastructure resources that are specific to the cluster and resources owned by the cluster will be created by the installer. So resources like security groups, roles, RHCOS boot images, and ignition storage objects.
+- The infrastructure resources owned by the cluster continue to be clearly identifiable and distinct from other resources.
+- Destroying a cluster must make sure that no resources are deleted that didn't belong to the cluster. Leaking resources is preferred over any possibility of deleting non-cluster resources.
+
+
+## User Stories
+1. As an administrator of a private openstack cloud, I use provider networks for my openstack cluster, and would like to be able to install OpenShift in this environment.
+2. As an OpenStack tenant, I want to be able to use pre-configured network resources so that I dont have to create new resources to install an OpenShift cluster.
+
+## Implementation Details Notes/Constraints
+
+### Installer Resources
+The user provides a list of subnets to the installer in the openstack platform section of the install-config.yaml.
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  openstack:
+    region: us-west-2
+    subnets:
+    - subnet-1
+    - subnet-2
+    - subnet-3
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+To make this work optimally, we should also enable customers to pass custom IP addresses to be used for our loadbalancing VIPs. This allows users who want to provision their own networks to set the CIDRs of the allowed address pairs that the VIPs can be drawn from.
```

Comment:
would like to see a section dedicate to this like the subnets https://github.com/openshift/enhancements/pull/246/files#diff-ca0ce402105d3b5a18fdcc4433cd013aR93

that details the validations/ requirements/ use etc. in detail.